main.go

```diff
@@ -81,77 +104,96 @@ func addrAllowed(addr string, allowedAddrs []string) bool {
 }
 
 func senderChecker(peer smtpd.Peer, addr string) error {
+	if *allowedSender == "" {
+		// disable sender check, allow anyone to send mail
+		return nil
+	}
```

Comment:
I didn't realize this during my first review, but moving this to the top of this function bypasses the per-user sender checks, and I disagree with this change.

However, perhaps what you really wanted was to keep user-auth, but allow them to send from any address. If so you're in luck, because that's exactly what I wanted too, and implemented last month in #7. The @grafana branch was just a little behind. Keeping up with upstream helps :grin: 

@electron0zero Can you confirm my understanding and that #7 supports your use-case?

Cross-ref: https://github.com/grafana/smtprelay/pull/4/files#r593928631